### PR TITLE
gpu-screen-recorder: unstable-2023-11-18 -> unstable-2024-05-21

### PR DESCRIPTION
**Pull request by id3v1669**
## Description of changes

Just a bump of a version to latest

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by id3v1669** [comment]
@ofborg build gpu-screen-recorder

**Comment by dec05eba** [comment]
libxfixes should be added as a new dependency (for cursor capture when recording a single window on x11).
Also would it be possible to add support for this as well? https://github.com/NixOS/nixpkgs/issues/275481 if cap_sys_admin+ep capability is added for gsr-kms-server then a popup asking for recording wont show up everytime you try to record on amd/intel or on wayland, which is useful for starting replay on system startup and it also makes it less annoying to use.
cap_sys_nice+ep capability can also be added on gpu-screen-recorder binary which improves screen recording performance when the game you are recording is drops below the recording video fps target, especially when you record with constant framerate mode.

**Comment by id3v1669** [comment]
> gpu-screen-recorder-gtk should also be updated at the same time to the latest version as it might not work correctly if it doesn't match the version of gpu-screen-recorder.
> 
> libxfixes should be added as a new dependency (for cursor capture when recording a single window on x11). Also would it be possible to add support for this as well? #275481 if cap_sys_admin+ep capability is added for gsr-kms-server then a popup asking for recording wont show up everytime you try to record on amd/intel or on wayland, which is useful for starting replay on system startup and it also makes it less annoying to use. cap_sys_nice+ep capability can also be added on gpu-screen-recorder binary which improves screen recording performance when the game you are recording is drops below the recording video fps target, especially when you record with constant framerate mode.

Done, added --suffix instead of prefix and added extra dependency.
Regarding `gpu-screen-recorder-gtk`, [commit](https://github.com/NixOS/nixpkgs/compare/master...id3v1669:nixpkgs:gpu-screen-recorder-gtk) is ready and I'll create pr for it as soon as this one is merged.

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 315118` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gpu-screen-recorder</li>
    <li>gpu-screen-recorder-gtk</li>
  </ul>
</details>

**Comment by keenanweaver** [comment]
>if cap_sys_admin+ep capability is added for gsr-kms-server then a popup asking for recording wont show up everytime you try to record on amd/intel or on wayland, which is useful for starting replay on system startup and it also makes it less annoying to use. cap_sys_nice+ep capability can also be added on gpu-screen-recorder binary which improves screen recording performance when the game you are recording is drops below the recording video fps target, especially when you record with constant framerate mode.

This'll require a separate module/option and not just editing this package. I do this in my config, which has what it would look like in nixpkgs: https://github.com/keenanweaver/nix-config/blob/main/modules/apps/gpu-screen-recorder/default.nix

(Also, I'm that guy that has been emailing you about the weird replay delays!)

**Review by keenanweaver — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/gpu-screen-recorder/default.nix
Comment (id3v1669) [reply pkgs/applications/video/gpu-screen-recorder/default.nix:57]: Why? As far as I see most packages follow that syntax as a standard to avoid duplication of `lib.xxx`

Comment (id3v1669) [reply pkgs/applications/video/gpu-screen-recorder/default.nix]: > It's apparently in bad form. I don't sway one way or the other, it's just most recent PRs I've seen/made they wanted this change done. https://github.com/NixOS/nixpkgs/pull/294383/files https://github.com/NixOS/rfcs/pull/120/files

Done.
But instead of `with lib.maintainers`, I used `lib.maintainers.xxx` as there is only one maintainer.

**Review by keenanweaver — APPROVED** [review]

**Pull request merged**
